### PR TITLE
Consolidate storage-related packages into pkg/storage

### PR DESCRIPTION
**Pull request by carolynvs**
# What does this change
This consolidates the pkg/credentials pkg/parameters, and pkg/sanitizer packages into the storage package.

This helps avoid a lot of circular dependencies that get introduced because these packages all rely on each other in
various ways (they are just views on top of the storage layer).

# What issue does it fix
I'm working on the data migration and keep running into circular dependencies. I don't think these packages really warrant being split up, and it will make it easier for us to move this code around.

# Notes for the reviewer

# Checklist
- [ ] Did you write tests?
- [x] Did you write documentation?
- [ ] Did you change porter.yaml or a storage document record? Update the corresponding schema file.
- [ ] If this is your first pull request, please add your name to the bottom of our [Contributors][contributors] list. Thank you for making Porter better! 🙇‍♀️

# Reviewer Checklist
* Comment with /azp run test-porter-release if a magefile or build script was modified
* Comment with /azp run porter-integration if it's a non-trivial PR

[contributors]: https://porter.sh/src/CONTRIBUTORS.md

### CODE REVIEW

**Comment by carolynvs** [comment]
I spoke with Yingrong and the sanitizer package may better fit in the pkg/secrets package.

**Comment by carolynvs** [comment]
@VinozzZ I checked and moving sanitizer into the secrets package creates a circular dependency between pkg/storage and pkg/secrets. I get that the sanitizer moves data into secrets, but I think of it as belonging with storage because it's sanitizing the data we store.

File: pkg/porter/list_test.go [hunk 1]
```diff
@@ -4,19 +4,19 @@ import (
...
 	"get.porter.sh/porter/pkg/secrets"
+	"get.porter.sh/porter/pkg/storage"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
 func TestNewDisplayInstallation(t *testing.T) {
 	t.Run("installation has been installed", func(t *testing.T) {
-		cp := claims.NewTestClaimProvider(t)
+		cp := storage.NewTestClaimProvider(t)
```
Comment (VinozzZ) [inline pkg/porter/list_test.go:16]: Do we want to also change the `NewTestClaimProvider` to `NewTestStorageProvider` ?

File: pkg/storage/claim_storage_provider.go [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
 
-import "context"
+import (
+	"context"
+)
 
-// Provider is an interface for interacting with Porter's claim data.
-type Provider interface {
+// ClaimProvider is an interface for interacting with Porter's claim data.
+type ClaimProvider interface {
```
Comment (VinozzZ) [inline pkg/storage/claim_storage_provider.go:8]: Do you mind to explain why we use `claim` here?

File: pkg/storage/credential_store.go [hunk 3]
```diff
@@ -1,32 +1,43 @@
...
-var _ Provider = &CredentialStore{}
+var _ CredentialSetProvider = &CredentialStore{}
 
 const (
 	CollectionCredentials = "credentials"
 )
 
-// CredentialStore provides access to credential sets by instantiating plugins that
-// implement CRUD storage.
+// CredentialStore provides access to work with Porter
```
Comment (VinozzZ) [inline pkg/storage/credential_store.go:21]: Maybe something is missing after the world `access` ?

File: pkg/storage/credential_store.go [hunk 4]
```diff
@@ -1,32 +1,43 @@
...
+// credential sets, usually referred to as "credentials" as a shorthand.
+//
+// Credential Sets define mappings from a credential needed by a bundle to where
+// to look for it when the bundle is run. For example: Bundle needs Azure
+// storage connection string and it should look for it in an environment
+// variable named `AZURE_STORATE_CONNECTION_STRING` or a key named `dev-conn`.
+//
+// Porter discourages storing the value of the credential directly, though it
+// it is possible. Instead Porter encourages the best practice of defining
```
Comment (VinozzZ) [inline pkg/storage/credential_store.go:30]:
```suggestion
// is possible. Instead Porter encourages the best practice of defining
```

File: pkg/storage/credential_store.go [hunk 5]
```diff
@@ -1,32 +1,43 @@
...
 const (
 	CollectionCredentials = "credentials"
 )
 
-// CredentialStore provides access to credential sets by instantiating plugins that
-// implement CRUD storage.
+// CredentialStore provides access to work with Porter
+// credential sets, usually referred to as "credentials" as a shorthand.
+//
+// Credential Sets define mappings from a credential needed by a bundle to where
```
Comment (VinozzZ) [inline pkg/storage/credential_store.go:24]: I wonder if this documentation applies better with the `CredentialSet` struct instead of here

**Review by VinozzZ — COMMENTED** [review]
LGTM, just some couple nit questions

File: pkg/porter/list_test.go (hunk 1 above)
Comment (carolynvs) [reply pkg/porter/list_test.go:16]: That is in the follow-up PR #2053

File: pkg/storage/claim_storage_provider.go (hunk 2 above)
Comment (VinozzZ) [reply pkg/storage/claim_storage_provider.go:8]: That makes perfect sense. I just wanted to make sure I understand the scope of this PR. Thanks for clarifying it!

**Pull request merged**